### ISSUE TITLE
Fix the versioning

### ISSUE THREAD

**Issue by danielchatfield** [jashkenas/underscore#1805]
1.7.0 introduced **loads** of breaking changes.

The number of dependant modules which are now broken as a result is huge, personally I think that 1.7.0 should be killed (removed from npm) and 2.0 released - the longer the delay the harder it will be to do this.

underscore.js is solely consumed via package managers that mandate the use of semver, you may personally not like semver but that is what is used by the installers to determine compatibility. Last time this was brought up you stated that if you used semver then we would be on underscore version 47 now - well that is much better than having broken code everywhere and lodash has managed to keep the version number below 4.0.0 without breaking everyone's code.

**Comment by hopeseekr** [comment]
I guess it's from my days of developing C++ libraries, but here is the system I've been using my entire career. I can see how this would really help the Underscore project and provide key information to the larger communities as to what to expect in the future, while not hampering the project's velocity of change much.

## Differential Versioning #

### PROGRAM_ERA
1. At least 20-25%+ of the code has changed since the first release of the last PROGRAM_ERA. 
2. There can be many new major features, 
3. Heavy refactoring is allowed, 
4. Dead code and deprecated functions from past MINOR_VERSIONs removed, etc., 
5. Possibly quite a huge number (and even severe) backward compatibility API breaks.
6. Other developers know that PROGRAM_ERA upgrades may and probably will require a lot of effort and testing to adopt in their own projects.
7. Specifically with first releases (e.g., X.0.0), edge case regressions *probably* exist, so the project is best avoided in use on high-requirement production systems until the next MAJOR_VERSION. (In Windows land, this is referred to as, "Wait until the first Service Pack.")

## MAJOR_VERSION
1. New features have been added.
2. New public interfaces (public API methods, etc.) have been added.
3. All previously unreleased code,  except trivial bug fixes, goes into **odd numbered** MAJOR_VERSIONs. Users and developers know that this is equivalent to a Testing/Bleeding edge release.
4. All previously released code and subsequent fixes can be released as **even numbered** MAJOR_VERSIONs, once the per-project test code coverage quota has been met and there are no open bug reports of High or higher.
5. There can be **slight** backward incompatible changes, highlighted in ChangeLog and release notes, either 
    * Tweaks to return types that do not critically break things [e.g., null to false].
    * Tweaks to handling edge cases (e.g., missing file returns false instead of null).
    * API breaks to fix security holes and/or extreme bugs. Requires a special PRIORITY UPGRADE release statement sent to every subscriber. (Proxy classes and other workarounds should be used when possible to avoid BC breaks).
6. Once a MAJOR_VERSION is released, it is immediately copied to a new **odd numbered version**.  All new code that are not bug fixes should go into this new **odd numbered** MAJOR_VERSION, making sure to also port over MINOR_VERSION bug fixes from the **even numbered** MAJOR_VERSION.

## MINOR_VERSION
1. Backwards compatible improvements, refactorizations, and bug fixes that are not expected to require triage or behavior changes from end users or developers.
2. A security hole or extreme bug has been fixed. Requires a special PRIORITY UPGRADE release statement sent to every subscriber.
3. Ideally, no changes to the usage or code documentation should be required.

## Version Selection

The end user public should only ever use **even numbered** MAJOR_VERSIONs. The end developer should stay on the latest **even numbered** MAJOR_VERSION, however, they are encouraged to use the latest **odd numbered** MAJOR_VERSION to ensure their app will work on the next stable release.

It is also helpful if technically sophisticated end users and programmer hobbyists use the latest **odd numbered** MAJOR_VERSION and submit bug reports when encountered.

Basically, all new underscore development would occur in **odd numbered** MAJOR_VERSIONs, and at a certain point at regular intervals, a few days would be taken out of the maintainers' schedules to release **even numbered** MAJOR_VERSION stable releases, which could then be adopted by the end-users with more ready zeal.

It would require, however, some more restraint on heavy backward incompatibility changes on **even numbered** MAJOR_VERSIONs, and absolutely no BC breaks on **even numbered** MINOR_VERSIONs.

**Comment by elgubenis** [comment]
(ノಠ益ಠ)ノ彡┻━┻

**Comment by bobmagicii** [comment]
if i saw a 0.x.0 update i sure as hell would be reading changelogs...

https://www.npmjs.com/package/npm-check

**Comment by ViniciusSasso** [comment]
Underscore and Backbone are so dead now...lol

**Comment by acstll** [comment]
Dead or alive they deserve respect.
